primitives/api/src/lib.rs

```diff
@@ -654,53 +659,19 @@ pub fn deserialize_runtime_api_info(bytes: [u8; RUNTIME_API_INFO_SIZE]) -> ([u8;
 	(id, version)
 }
 
-#[derive(codec::Encode, codec::Decode)]
-pub struct OldRuntimeVersion {
-	pub spec_name: RuntimeString,
-	pub impl_name: RuntimeString,
-	pub authoring_version: u32,
-	pub spec_version: u32,
-	pub impl_version: u32,
-	pub apis: ApisVec,
-}
-
-impl From<OldRuntimeVersion> for RuntimeVersion {
-	fn from(x: OldRuntimeVersion) -> Self {
-		Self {
-			spec_name: x.spec_name,
-			impl_name: x.impl_name,
-			authoring_version: x.authoring_version,
-			spec_version: x.spec_version,
-			impl_version: x.impl_version,
-			apis: x.apis,
-			transaction_version: 1,
-		}
-	}
-}
-
-impl From<RuntimeVersion> for OldRuntimeVersion {
-	fn from(x: RuntimeVersion) -> Self {
-		Self {
-			spec_name: x.spec_name,
-			impl_name: x.impl_name,
-			authoring_version: x.authoring_version,
-			spec_version: x.spec_version,
-			impl_version: x.impl_version,
-			apis: x.apis,
-		}
-	}
-}
-
 decl_runtime_apis! {
 	/// The `Core` runtime api that every Substrate runtime needs to implement.
 	#[core_trait]
-	#[api_version(3)]
+	#[api_version(4)]
 	pub trait Core {
 		/// Returns the version of the runtime.
 		fn version() -> RuntimeVersion;
 		/// Returns the version of the runtime.
+		#[changed_in(4)]
+		fn version() -> RuntimeVersionV1;
+		/// Returns the version of the runtime.
 		#[changed_in(3)]
-		fn version() -> OldRuntimeVersion;
+		fn version() -> RuntimeVersionV0;
```

Comment:
If we make `RuntimeVersion` decode being able to decode from V0 to V2, we don't need to do any of these `changed_in` attributes here. We can just have one `fn version() -> RuntimeVersion` and that would be enough.

Reply:
Nice, I am not too familiar with these `changed_in` purpose, will do the switch.

Reply:
https://github.com/paritytech/substrate/pull/9732/commits/cf875e05076502b792e355fc61b2feb07fcd1f1a does it.
@bckhr, I also realize that sometime runtime version do not read the core api version, which make this commit a bit dangerous. I'm considering reverting it.

Reply:
What you mean by this?

Reply:
The default decode implementation of `RuntimeVersion` that expects apis versions to be read (which is not always the case).